services/state/schedule/schedule.proto

```diff
@@ -54,54 +57,64 @@ message Schedule {
     string memo = 3;
 
     /**
-     * This schedule's unique ID within the network state.
+     * This schedule's unique ID within the global network state.
      */
     ScheduleID id = 4;
 
     /**
-     * The schedule account for this schedule.
+     * The schedule account for this schedule.  This is the account that submitted the original
+     * ScheduleCreate transaction.
      */
     AccountID scheduler_account = 5;
 
     /**
-     * The payer account for this schedule.
+     * The explicit payer account for the scheduled transaction.
+     * This account is added to the accounts that must sign the schedule before it will execute.
      */
     AccountID payer_account = 6;
 
     /**
-      * The admin key for this schedule.
+     * The admin key for this schedule.
+     * If this is not set, then the schedule cannot be deleted.
      */
     Key admin_key = 7;
 
     /**
-      * The start time of the schedule.
+     * The transaction valid start value from the transaction that created this schedule.
      */
     Timestamp schedule_valid_start = 8;
 
     /**
-      * The expiration time of the schedule as provided by the user.
+     * The requested expiration time of the schedule as provided by the user.
+     * The actual calculated expiration time may be "earlier" than this, but will not be later.
      */
     Timestamp expiration_time_provided = 9;
 
     /**
-      * The calculated expiration time of the schedule.
+     * The calculated expiration time of the schedule.  This is calculated based on the requested
+     * expiration time from the create transaction, and the maximum expiration permitted by the
+     * network.
+     * The schedule will be removed from global network state after the network reaches a consensus
+     * time greater than or equal to this value.
      */
     Timestamp calculated_expiration_time = 10;
 
     /**
-      * The consensus timestamp of the transaction that executed, deleted, or expired this schedule.
+     * The consensus timestamp of the transaction that executed or deleted this schedule.
      */
     Timestamp resolution_time = 11;
 
     /**
-     * The scheduled transaction
+     * The scheduled transaction to execute.
      */
     SchedulableTransactionBody scheduled_transaction = 12;
 
     /**
-     * The transaction body bytes
+     * The full transaction that created this schedule.  This is primarily used for duplicate
+     * schedule create detection.  This is also the source of the parent transaction ID, from
+     * which the child transaction ID is derived.
      */
-    bytes body_bytes = 13;
+    TransactionBody original_create_transaction = 13;
```

Comment:
This is probably a dumb question, but is it ok to change the type on this if we need to be backwards compatible?

Reply:
The only code that uses this is on my branch and not merged, and nothing has stored any of these objects in state, so we don't need backwards compatibility yet.
Once that code merges and there is state with these values, we will have to be immensely more careful, hence the work to make these as correct as possible at the start.

Reply:
Sounds good, thanks for clarifying!